File: backend/spring/pulse/src/main/java/dev/thilanka/netrics/service/MapCellThrSetService.java
package dev.thilanka.netrics.service;

import dev.thilanka.netrics.dto.MapCellThrSetDto;
import dev.thilanka.netrics.dto.MapCellThrSetResponseDto;
import dev.thilanka.netrics.entity.Granularity;
import dev.thilanka.netrics.entity.MapCellThrSet;
import dev.thilanka.netrics.entity.Rat;
import dev.thilanka.netrics.entity.StandardKpi;

public interface MapCellThrSetService {

    MapCellThrSet createThrSet(MapCellThrSet thrSet);

    MapCellThrSet createThrSet(MapCellThrSetResponseDto thrSetDto);

    MapCellThrSetResponseDto createThrSetResponse(MapCellThrSetResponseDto dto);

    MapCellThrSet findThrSetById(Long id);

    MapCellThrSet findThrSet(StandardKpi standardKpi, Rat rat, Granularity granularity, boolean isAdmin);

    MapCellThrSetDto getThrSet(String standardKpiName, String ratName, String granularityName, boolean isAdmin);

    MapCellThrSetResponseDto getThrSetResponse(String standardKpiName, String ratName, String granularityName, boolean isAdmin);
}

File: frontend/netrics/src/app/service/pulse/map-cell/map-cell-thr-set-service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {UrlService} from '../../url/url-service';
import {MapCellThrSetResponseDto} from '../../../models/pulse/map-cell/MapCellThrSetResponseDto';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class MapCellThrSetService {

  private readonly baseUrl: string;

  constructor(private http: HttpClient, private urlService: UrlService) {
    this.baseUrl = `${this.urlService.getPulseUrl()}/map-cell-thr-sets`;
  }

  createThrSet(thrSet: MapCellThrSetResponseDto): Observable<MapCellThrSetResponseDto> {
    return this.http.post<MapCellThrSetResponseDto>(`${this.baseUrl}`, thrSet)
  }

  getThrSetResponse(standardKpiName: string, ratName: string, granularityName: string, isAdmin: boolean) {
    const params = new HttpParams()
      .set('standardKpiName', standardKpiName)
      .set('ratName', ratName)
      .set('granularityName', granularityName)
      .set('isAdmin', isAdmin);

    return this.http.get<MapCellThrSetResponseDto>(`${this.baseUrl}`,{params})
  }

}
